dialog/color.go

```diff
@@ -49,6 +49,9 @@ func (p *ColorPickerDialog) SetColor(c color.Color) {
 
 // Refresh causes this dialog to be updated
 func (p *ColorPickerDialog) Refresh() {
+	if w := p.win; w != nil {
+		w.Hide()
```

Comment:
This seems strange... It will hide an existing dialog if you call Refresh() - but I don't see why.
a) there is no Show() at the end of this method
b) can't we change the content of the dialog without hiding it first?

Reply:
The problem is that Show doesn't create the popup, setButtons does

Reply:
Sure, but setButtons could be called later right?

Reply:
Where should it be called?

Reply:
My immediate thought would be that it's set the first time Show is called - would that work OK?
After first Show then Refresh would clearly be needed.

Reply:
Refresh and Show now both delegate to a helper, does this resolve your concern or have I misunderstood?

Reply:
That seems good